chapter/related-work.md

```diff
@@ -6,6 +6,90 @@ As such, it is important to get a picture of the field of current LSP projects.
+Additional languages can be integrated by custom (third-party) plugins or derivations of the base platform ([@list-of-eclipse,@jetbrains-all-products]).
+Due to the proprietary nature of some of these products, plugins are not compatible between different platforms.
+Many less popular languages therefore saw redundant implementations of what is essentially the same.
+For Haskell separate efforts produced an eclipse based IDE [@haskell-ide-eclips], as well as independent IntelliJ plugins [@intellij-haskell,@HaskForce].
+Importantly, the implementers of the former reported troubles with the language barrier between Haskell and the Eclipse base written in Java.
+
+The Haskell language is an exceptional example since there is also a native Haskell IDE[@haskell-for-mac] albeit that it is available only to the MacOS operating system.
+This showcases the difficulties of language tooling and its provision.
+
+In general, developing language integrations, both as the vendor of an IDE or third-party plugin developer requires extensive resources.
+[Table @tbl:plugins-size] gives an idea of the efforts-required.
+Strikingly, since the IntelliJ platform is based on the JVM, its plugin system requires the use of JVM languages [@custom-language-support]
+The Rust and Haskell integrations for instance contain at best only a fraction of code in their respective language.
+
+| Plugin                    | lines of code                                                |
+| ------------------------- | ------------------------------------------------------------ |
+| intellij-haskell          | 17249 (Java) + 13476 (Scala) **+ 0 (Haskell)**               |
+| intellij-rust             | 229131 (Kotlin) + 3958 (Rust)                                |
+| intellij-scala            | 39382 (Java) + 478904 (Scala)                                |
+| intellij-kotlin           | 182372 (Java) + 563394 (Kotlin)                              |
+| intellij-community/python | 47720 (C) + 248177 (Java) + 37101 (Kotlin) + 277125 (Python) |
+
+: Comparison of the size for different IntelliJ platform plugins {#tbl:plugins-size}
+
+Naturally, development efforts at this size would gravitate around the most promising solution, stifling the progress on competing platforms [@intellij-comparison-eclipse].
+Additionally, it would lock-in programmers into a specific platform for its language support regardless of their personal preference.
+
+### IDE Abstraction
+
+#### Monto
+
+The authors of the Monto project[@monto,@monto-disintegrated] call this the "IDE Portability Problem".
+They compare the situation with the task of compiling different high level languages to a set of CPU architectures.
+The answer to that problem was an intermediate representation (IR).
+Compilers could transform input languages into this IR and in turn generate assembly for different architectures from a single input format.
```

Comment:
This sentence is a bit confusing because it sounds like you are describing Monto, which appears as an IR for code generation. Maybe a comparison would make things clearer, and I guess LLVM is a good comparison.